src/lib/pages/backoffice/Loan/LoanDetails/AvailableItems/AvailableItems.js

```diff
@@ -21,12 +21,18 @@ export default class AvailableItems extends Component {
     this.assignItemToLoan = props.assignItemToLoan;
     this.performCheckoutAction = props.performCheckoutAction;
     this.seeAllUrl = BackOfficeRoutes.itemsListWithQuery;
-    this.state = { query: '', filteredData: null };
+    this.state = {
+      query: '',
+      filteredData: null,
+      modalOpen: false,
+      checkoutItem: null,
+    };
   }
 
   componentDidMount() {
-    const { fetchAvailableItems, loan } = this.props;
+    const { fetchAvailableItems, fetchPendingLoans, loan } = this.props;
     fetchAvailableItems(loan.metadata.document_pid);
+    fetchPendingLoans(loan.metadata.document_pid);
```

Comment:
It is safer to chain the requests to avoid issues a multiple re-renderings. I think that to achieve this, you will have to modify the `fetchAvailableItems` (if used only here!!!), e.g. `fetchAvailableItemsWithPendingLoans`, and inside the action.js:
```
await itemApi.list(
        itemApi
          .query()
          .withDocPid(documentPid)
          .withStatus(invenioConfig.ITEMS.canCirculateStatuses)
          .availableForCheckout()
          .qs()
      );

await loans...

update Redux with the info you need, for example, for each item:

item["is_first_requested"] = True/False
```
and then use that `is_first_requested` to display the modal or not.

Reply:
I understand your point, just to clarify this, we need to modify the `fetchLoanDetails` instead the `fetchAvailableItems`, since we are searching for the first requested **loan**. Thanks !